utils/staking-miner/src/main.rs

```diff
@@ -0,0 +1,517 @@
+			let sum_squared = winners.iter().fold(0u128, |acc, (_, stake)| acc + stake);
+			[min_staker, sum_stake, sum_squared]
+		};
+		println!("mined a dpos-like solution with score = {:?}", score);
+		Ok(())
+	})
+}
+
+pub(crate) async fn get_account_info<T: frame_system::Config>(
+	client: &WsClient,
+	who: &T::AccountId,
+	maybe_at: Option<T::Hash>,
+) -> Result<Option<frame_system::AccountInfo<Index, T::AccountData>>, Error> {
+	rpc_helpers::get_storage::<frame_system::AccountInfo<Index, T::AccountData>>(
+		client,
+		params! {
+			sp_core::storage::StorageKey(<frame_system::Account<T>>::hashed_key_for(&who)),
+			maybe_at
+		},
+	)
+	.await
+}
+
+/// Read the signer account's uri from the given `path`.
+async fn read_signer_uri<
+	P: AsRef<Path>,
+	T: frame_system::Config<AccountId = AccountId, Index = Index>,
+>(
+	path: P,
+	client: &WsClient,
+) -> Result<Signer, Error> {
+	let uri = std::fs::read_to_string(path)?;
+
+	// trim any trailing garbage.
+	let uri = uri.trim_end();
+
+	let pair = Pair::from_string(&uri, None)?;
+	let account = T::AccountId::from(pair.public());
+	let _info =
+		get_account_info::<T>(&client, &account, None).await?.ok_or(Error::AccountDoesNotExists)?;
+	log::info!(target: LOG_TARGET, "loaded account {:?}, info: {:?}", &account, _info);
+	Ok(Signer { account, pair, uri: uri.to_string() })
+}
+
+#[tokio::main]
+async fn main() {
+	env_logger::Builder::from_default_env().format_module_path(true).format_level(true).init();
+	let Opt { shared, command } = Opt::from_args();
+	log::debug!(target: LOG_TARGET, "attempting to connect to {:?}", shared.uri);
+
+	let client = WsClientBuilder::default()
+		.connection_timeout(std::time::Duration::new(20, 0))
+		.max_request_body_size(u32::MAX)
+		.build(&shared.uri)
+		.await
+		.unwrap();
+
+	let chain = rpc_helpers::rpc::<String>(&client, "system_chain", params! {})
+		.await
+		.expect("system_chain infallible; qed.");
+	match chain.to_lowercase().as_str() {
+		"polkadot" | "development" => {
+			sp_core::crypto::set_default_ss58_version(
+				sp_core::crypto::Ss58AddressFormat::PolkadotAccount,
+			);
+			// safety: this program will always be single threaded, thus accessing global static is
+			// safe.
+			unsafe {
+				RUNTIME = AnyRuntime::Polkadot;
+			}
+		}
+		"kusama" | "kusama-dev" => {
+			sp_core::crypto::set_default_ss58_version(
+				sp_core::crypto::Ss58AddressFormat::KusamaAccount,
+			);
+			// safety: this program will always be single threaded, thus accessing global static is
+			// safe.
+			unsafe {
+				RUNTIME = AnyRuntime::Kusama;
+			}
+		}
+		"westend" => {
+			sp_core::crypto::set_default_ss58_version(
+				sp_core::crypto::Ss58AddressFormat::PolkadotAccount,
+			);
+			// safety: this program will always be single threaded, thus accessing global static is
+			// safe.
+			unsafe {
+				RUNTIME = AnyRuntime::Westend;
+			}
+		}
+		_ => panic!("unexpected chain: {:?}", chain),
```

Comment:
```suggestion
		_ => {
			eprintln!("unexpected chain: {:?}", chain);
			return;
		}
```

Panicing produces a bunch of other debug info that the user doesn't need or want in this case. We could also get fancy and `std::process::exit(1)`, but I'm not sure if we'd need a wrapper function in this case to ensure that everything gets dropped appropriately.